### PR TITLE
Refocus config as args

### PR DESCRIPTION
**Pull request by iamigo**

### CODE REVIEW

**Comment by shriramshankar** [comment]
@iamigo Looks like there are a lot of merge conflicts.

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,4328 @@
+{
```
Comment (shriramshankar) [inline package-lock.json]: Are we using npm ci for the collectors too?

Comment (iamigo) [reply package-lock.json]: I added the package lock so we could set up npm ci if we wanted to

Comment (iamigo) [reply package-lock.json]: happy to remove it though, if we're not ready to switch over to that yet

**Pull request merged**